File: 56-merge-intervals/merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
         if(intervals.size()==1) return intervals; 
         sort(begin(intervals),end(intervals),[](const vector<int> &a, const vector<int> &b) {
            return a[0]<b[0];
         });  int start=intervals[0][0];
        int end=intervals[0][1];
    
        vector<vector<int>> ans;
        for(int i=1; i<intervals.size(); i++) {
         
             if(intervals[i][0]<=end) {
              end=max(intervals[i][1],end);
             }
             else {
                ans.push_back({start,end});
               
                    start=intervals[i][0];
                    end=intervals[i][1];
                   
                
             } if(i+1 == intervals.size()) {
                        ans.push_back({start,end});
                    }

        }
        return ans;
    }
};

File: 56-merge-intervals/merge-intervals.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function(intervals) {
    intervals.sort((a,b) => {
        if(a[0]==b[0]) return a[1]-b[1];

        return a[0]-b[0];
    });
   let end=intervals[0][1];
   let start=intervals[0][0];
   let temp=[];
    for(let i=1; i<intervals.length; i++) {
            if(intervals[i][0]>end) {
                 temp.push([start,end]);
                 start=intervals[i][0];
                 end=intervals[i][1];
            }
            end=Math.max(end,intervals[i][1]);  
    }
    temp.push([start,end]);

    return temp;
};